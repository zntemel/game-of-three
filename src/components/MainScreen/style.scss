.main-screen {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  padding: 50px;
  .game-name {
    font-weight: bold;
    font-size: 2em;
    margin-bottom: 0;
  }
  .main-screen-divider {
    color: #e7e7f3;
    margin-bottom: 60px;
  }
  button {
    margin-bottom: 40px;
    height: 60px;
    border: none;
    padding: 15px 50px;
    min-width: 60%;
    font-size: 16px;
  }
  .copyright {
    color: #000
  }
}
