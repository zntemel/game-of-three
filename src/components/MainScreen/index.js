import React from "react";
import { Divider, Button } from "antd";
import "./style.scss";

function MainScreen(props) {
  return (
    <div className="main-screen">
      <h1 className="game-name">Game Of Three</h1>
      <Divider className="main-screen-divider" />
      <Button onClick={() => props.history.push("/singleplayer")}>
        Play With Computer
      </Button>
      <Button onClick={() => props.history.push("/multiplayer")}>
        Play With a Real Player
      </Button>
      <p className="copyright">Copyright Â©2020</p>
    </div>
  );
}

export default MainScreen;
