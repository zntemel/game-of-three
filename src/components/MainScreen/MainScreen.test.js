import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import MainScreen from "./index";

describe("MainScreen", () => {
  const wrapper = shallow(<MainScreen />);
  const instance = wrapper.instance();
  const gameName = "Game Of Three";

  describe("shallow wrapper", () => {
    it("should render succesfully with snapshots", () => {
      const component = renderer.create(wrapper).toJSON();
      expect(component).toMatchSnapshot();
    });

    it("instance should be null because of it's dumy component", () => {
      expect(instance).toEqual(null);
    });

    it("should be include name game", () => {
      expect(wrapper.find("h1").contains(gameName)).toBe(true);
    });

    it("should be include divider component", () => {
      expect(wrapper.find("Divider").exists()).toBeTruthy();
    });

    it("should be include two button", () => {
      expect(wrapper.find("Button")).toHaveLength(2);
    });
  });
});
