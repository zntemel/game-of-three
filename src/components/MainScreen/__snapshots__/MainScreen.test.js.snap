// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen shallow wrapper should render succesfully with snapshots 1`] = `
<div
  className="main-screen"
>
  <h1
    className="game-name"
  >
    Game Of Three
  </h1>
  <div
    className="main-screen-divider ant-divider ant-divider-horizontal"
    role="separator"
  />
  <button
    className="ant-btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Play With Computer
    </span>
  </button>
  <button
    className="ant-btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Play With a Real Player
    </span>
  </button>
  <p
    className="copyright"
  >
    Copyright Â©2020
  </p>
</div>
`;
